### PR TITLE
Add validate contact file to scan pipe.

### PR DESCRIPTION
**Pull request by datamel**
These changes close https://github.com/cylc/cylc-uiserver/issues/343

Sibling pr: https://github.com/cylc/cylc-flow/pull/4857

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests (adapted).
- [x] No change log entry required (why? invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Comment by datamel** [comment]
Broken tests and need to add tests ui server end. Will fix when I'm back.

**Comment by datamel** [comment]
Tests will fail unless run with https://github.com/cylc/cylc-flow/pull/4857.

**Review by hjoliver — APPROVED** [review]

File: cylc/uiserver/workflows_mgr.py [hunk 1]
```diff
@@ -141,14 +142,16 @@ def __init__(self, uiserver, log, context=None, run_dir=None) -> None:
             | is_active(is_active=True, filter_stop=False)
             # extract info from the contact file
             | contact_info
+            # ensure contact file is not corrupt
```
Comment (oliver-sanders) [inline cylc/uiserver/workflows_mgr.py:145]:
```suggestion
            # ensure required contact file fields are present
```

**Review by oliver-sanders — APPROVED** [review]
LGTM :+1:

**Comment by hjoliver** [comment]
Sibling cylc-flow PR merged.

**Comment by hjoliver** [comment]
Tests passed after reboot :taco:

**Pull request merged**